* Only "pop" subtree params if they had previously been "pushed", and 
  afterwards reset the "pushed" value to 0 again. Similarly only set the
  PLAY_TREE_RND_PLAYED flag if the entry had been pushed before.
  OBSOLETE: remove this patch at next MPlayer upgrade.
diff -Naur MPlayer-export-2009-06-13.orig/playtree.c MPlayer-export-2009-06-13/playtree.c
--- MPlayer-export-2009-06-13.orig/playtree.c	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/playtree.c	2009-06-13 22:54:13.000000000 +0200
@@ -712,8 +712,9 @@
   iter->tree = iter->tree->parent;
 
   // Pop subtree params
-  if(iter->config) {
+  if(iter->config && iter->entry_pushed > 0) {
     m_config_pop(iter->config);
+    iter->entry_pushed = 0;
     if(iter->mode == PLAY_TREE_ITER_RND)
       iter->tree->flags |= PLAY_TREE_RND_PLAYED;
   }
